CASSANDRA-18244 Improve documentation about releases for Cassandra website (#208)
apache/cassandra-website: site-content/source/modules/ROOT/pages/development/release_process.adoc

----

Publish your GPG key in a PGP key server, such as
http://pgp.mit.edu/[MIT Keyserver]. Some `gpg` clients are publishing the keys https://keys.openpgp.org/[here]. You are 

> welcome to set the server the keys will be published to by following https://keys.openpgp.org/about/usage[this guide].

Reviewer: ```suggestion
welcome to set the server where the keys will be published by following https://keys.openpgp.org/about/usage[this guide].
```

> Once done, you need to create a ticket https://issues.apache.org/jira/browse/CASSANDRA-18205[like this] 

Reviewer: ```suggestion
Once completed, you need to create a ticket https://issues.apache.org/jira/browse/CASSANDRA-18205[like this] 
```

and ask a PMC to add your key to `KEYS` file.

=== Create Release Artifacts

Any committer can perform the following steps to create and call a vote
on a proposed release.

Check that there are no open urgent Jira tickets currently being worked
on. Also check with the PMC that there's security vulnerabilities
currently being worked on in private. Current project habit is to check
the timing for a new release on the dev mailing lists.

> For sucessful building process, you need to have this tooling installed locally: 

Reviewer: ```suggestion
For successful building process, install this tooling locally: 
```

svn, git, ant, devscripts, reprepro, rpmsign, docker, createrepo (the script is checking this tooling is present 

> before proceeding any further). The names of these "packages" are Debian-centric but there 

Reviewer: ```suggestion
before proceeding any further). The names of these "packages" are Debian-centric, 
```

> should not be any problem to find the equivalents in other systems too.

Reviewer: ```suggestion
but equivalents should be discoverable in other systems too.
```

cd ~/git
git clone https://github.com/apache/cassandra-builds.git
git clone https://github.com/apache/cassandra.git
----

[source,none]
----
# Edit the variables at the top of the `prepare_release.sh` file
edit cassandra-builds/cassandra-release/prepare_release.sh
----

> You need to specify your ASF username to `asf_username` variable. Next, `gpg_key` environment variable needs to be 

Reviewer: ```suggestion
You must specify your ASF username to `asf_username` variable. Next, `gpg_key` environment variable must be 
```

> set to a fingerprint of your gpg key. You get that value by executing `gpg --list-keys` or similar. Also, you need to 

Reviewer: ```suggestion
set to a fingerprint of your gpg key. Execute `gpg --list-keys` or a similar command to get the value. Finally, you must
```

> add ASF remote to cloned repository and `git_asf_remote` variable needs to be set to point to that. For example, when this is done:

Reviewer: ```suggestion
add ASF remote to your cloned repository and `git_asf_remote` variable needs to be set to point to that. For example, when this command is executed:
```

[source,none]
----
git remote add asf https://gitbox.apache.org/repos/asf/cassandra.git
----

> then `git_asf_remote` variable needs to be set to `asf`. This is very important step as tags are pushed to ASF repository and 

Reviewer: ```suggestion
then `git_asf_remote` variable needs to be set to `asf`. 
NOTE: This is very important step as tags are pushed to ASF repository and they are synchronized to GitHub automatically.
```

> they are synchronized to GitHub automatically.

Reviewer: ```suggestion
```

[source,none]
----
# Ensure your 4096 RSA key is the default secret key
edit ~/.gnupg/gpg.conf # update the `default-key` line
----

> A reference configuration looks like this:

Reviewer: ```suggestion
A reference configuration should look like these examples:
```

[source,none]
----
default-key <fingerprint of your key>
personal-digest-preferences SHA512
cert-digest-algo SHA512
default-preference-list SHA512 SHA384 SHA256 SHA224 AES256 AES192 AES CAST5 ZLIB BZIP2 ZIP Uncompressed
----

[source,none]
----
edit ~/.rpmmacros # update the `%_gpg_name <key_id>` line
# Ensure DEBFULLNAME and DEBEMAIL is defined and exported, in the debian scripts configuration
edit ~/.devscripts
----

> The reference content of these files should be:

Reviewer: ```suggestion
The reference content of these files is:
```

[source,none]
----
$ cat ~/.rpmmacros 
%_gpg_name email@youusedforyourkey.org
$ cat ~/.devscripts 
DEBFULLNAME="Your Name"
DEBEMAIL=email@youusedforyourkey.org
----

> It was empirically tested that you also need to have the above `DEB*` environemnt variables exported before proceeding.

Reviewer: ```suggestion
Empirically testing shows that you also must have the above `DEB*` environemnt variables exported before proceeding.
```

> Empirically testing shows that you also must have the above `DEB*` environemnt variables exported before proceeding.

Reviewer 2: ```suggestion
Empirical testing shows that you also must have the above `DEB*` environemnt variables exported before proceeding.
```

> Next, you have to configure `$HOME/.m2/settings.xml` to contain the credentials how to upload artifacts to https://repository.apache.org/[staging repository]. The credentails are your ASF credentials.

Reviewer: ```suggestion
Additionally, you must configure `$HOME/.m2/settings.xml` to contain the credentials used to upload artifacts to https://repository.apache.org/[staging repository]. The credentials are your ASF credentials.
```

Follow the prompts.

> Once artifacts are built and pushed to the staging area, the script will pause and it will ask you to go to

Reviewer: ```suggestion
Once artifacts are built and pushed to the staging area, the script will pause and require you to go to the
```

> https://repository.apache.org/#stagingRepositories[Staging repositories] and you will find the repository there. 

Reviewer: ```suggestion
https://repository.apache.org/#stagingRepositories[Staging repositories], where you will find the repository. 
```

> You need to select Cassandra repository and push the "Close" button. 

Reviewer: ```suggestion
Select the Cassandra repository and push the "Close" button. 
```

> Please take a note of a number of that repository like `orgapachecassandra-1283` - number is 1283. 

Reviewer: ```suggestion
Please take a note of the number of that repository like `orgapachecassandra-1283` - number is 1283. 
```

> This will be asked by the script to proceed. It will use this number upon rendering the e-mail template to send to dev list etc.

Reviewer: ```suggestion
The script will require the repo number to proceed. It will use this number in the rendered e-mail template sent to the dev list, etc.
```

If building the deb or rpm packages fail, those steps can be repeated
individually using the [.title-ref]#-d# and [.title-ref]#-r# flags,
respectively.

> Once DEBs and RPMs are uploaded too, do not forget to merge up your commit which prepares the release up to trunk 

Reviewer: ```suggestion
Once DEBs and RPMs are also uploaded, do not forget to merge your commit to prepare the release to trunk 
```

> and push after all artifacts are uploaded and whole process is finished. You will be also reminded to do that by 

Reviewer: ```suggestion
and push after all artifacts are uploaded finish the process. You will be also reminded to do that by 
```
